### PR TITLE
docs(browse): add documentation on browse feature

### PR DESCRIPTION
**Pull request by RyanHolstien**
## Checklist
- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: docs/browse.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+</p>
+
+## Additional Resources
+
+### GraphQL
+
+* [browse](../../graphql/queries.md#browse)
+* [browsePaths](../../graphql/queries.md#browsePaths)
+
+## FAQ and Troubleshooting
```
Comment (maggiehays) [inline docs/browse.md:31]: Let's address these common questions:

* How are browse paths created?
* How can I customize browse paths?

File: docs/browse.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
+import FeatureAvailability from '@site/src/components/FeatureAvailability';
+
+# About DataHub Browse
+
+<FeatureAvailability/>
+
+Browse is one of the primary entrypoints for discovering different Datasets, Dashboards, Charts and other DataHub Entities.
+
+Browsing is useful for finding data based on a hierarchical structure. A user might want to find all of their Datasets of a particular platform
+under a particular naming prefix for an organization, for example finding all Snowflake Datasets under the long_tail_companion organization organized under the analytics prefix.
```
Comment (maggiehays) [inline docs/browse.md:10]: Let's be a bit more explicit here --

> Browsing is useful for finding data entities based on a hierarchical structure set in the source system. Generally speaking, that hierarchy will contain the following levels:
> 
> * Entity Type (Dataset, Dashboard, Chart, etc.)
> * Environment (prod vs. dev)
> * Platform Type (Snowflake, dbt, Looker, etc.)
> * Container (Warehouse, Schema, Folder, etc.)
> * Entity Name
> 
> For example, a user can easily browse for Datasets within the PROD Snowflake environment, the  `long_tail_companions` warehouse, and the  `analytics` schema: 

datahub_browse

File: docs/browse.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+## Using Browse
+
+Browse is accessible by clicking into an entity on the front page of the DataHub UI.
```
Comment (maggiehays) [inline docs/browse.md:14]: > Browse is accessible by clicking on an Entity Type on the front page of the DataHub UI.

**Review by maggiehays — CHANGES_REQUESTED** [review]

File: docs/browse.md (hunk 2 above)
Comment (RyanHolstien) [reply docs/browse.md]: If you can send me this video or add it to the static assets I can add the link to it as well

**Review by maggiehays — APPROVED** [review]

**Pull request merged**